README.md

```diff
@@ -1,38 +1,275 @@
 # Sparrow
+**Summary**
+Sparrow is Elixir library for [push notifications](en.wikipedia.org/wiki/Push_technology#Push_notification).
+[![Build Status](https://travis-ci.org/esl/sparrow.svg?branch=master)](https://travis-ci.org/esl/sparrow)
+[![Coverage Status](https://coveralls.io/repos/github/esl/sparrow/badge.svg)](https://coveralls.io/github/esl/sparrow)
 
-**TODO: Add description**
+Currently provides support for following APIs:
+* [FCM v1](https://firebase.google.com/docs/cloud-messaging/)
+* [APNS](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html#//apple_ref/doc/uid/TP40008194-CH8-SW1)
 
-**TODO: Redirect Erlang lager to Elixir logger**
+**Requirements**
+* Elixir 1.6 or higher
+* Erlang OTP 20.3.6 or OTP 21
 
-**TODO: Add publishing package to hex** 
+# Build sparrow config
+This section describes how to write config file for Sparrow.
+If you wish to use just one of the following services do not include the other in config.
 
-**TODO: Check API diffs between these articles:** 
+```elixir
+config :sparrow, config: [fcm_config, apns_config]
+```
+**Config opts**
 
-[link](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/generating_a_remote_notification)
-and
-[link](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW1)
+***FCM specific***
+- `path_to_json` - is a path to json file provided by FCM
 
-## Installation
+***APNS specific***
+- `auth_type` - defines a authentocation type, allowied values are: `:token_based`, `:certificate_based`
+    1 `:token_based` requres to set:
+    - `token_id` - is a unique atom refering to (the same atom) token in tokens in config
+    
+    2 `:certificate_based` requres to set:
+    
+    - `cert` - path to certificate file provided by APNS
+    - `key` - path to key file provided by APNS
 
-[![Build Status](https://travis-ci.com/esl/sparrow.svg?branch=master)](https://travis-ci.com/esl/sparrow)
-[![Coverage Status](https://coveralls.io/repos/github/esl/sparrow/badge.svg)](https://coveralls.io/github/esl/sparrow)
+***H2Worker config***
+- `endpoint` - service uri address 
+- `port` - service port
+- `tls_opts` - passed to erlang [ssl](http://erlang.org/doc/man/ssl.html) module (see DATA TYPES -> ssl_option())
+- `ping_interval` - after each `ping_interval` miliseconds [ping frame](https://http2.github.io/http2-spec/#PING) is send through connection
```

Comment:
number of miliseconds between each [ping](https://http2.github.io/http2-spec/#PING)